numpy/core/src/multiarray/descriptor.c

```diff
@@ -3408,10 +3417,15 @@ NPY_NO_EXPORT PyTypeObject PyArrayDescr_Type = {
     0,                                          /* tp_hash */
     0,                                          /* tp_call */
     (reprfunc)arraydescr_str,                   /* tp_str */
-    0,                                          /* tp_getattro */
-    0,                                          /* tp_setattro */
+    PyObject_GenericGetAttr,                    /* tp_getattro */
+    PyObject_GenericSetAttr,                    /* tp_setattro */
```

Comment:
This is only necessary because of the `memset` you're doing above that copies `dtype.__dict__` to `dtype(int).__dict__`, something you really shouldn't be doing,

Reply:
I'm being dumb, and misread `memset` as the early `memcpy`. I'm still surprised that you need to do this though.

Reply:
Me too. I am still trying to work out the subtleties
- inheriting from type
- what is the deeper reason I need to begin the `PyArray_Descr` struct with a `PyTypeObject`, even before I begin implementing `np.dtype(int)()` or subclassing

Maybe this would be easier to reason about in pure python

Reply:
> Maybe this would be easier to reason about in pure python

In pure python, if one subclasses `type`, at some point one would do `super().__new__` and `super().__init__`. Do you do that here? In particular, I noticed you introduced a a C version of `__init__` which just does `return 0`; should this perhaps instead replicate what `type` would do? Might this inherited function have been removed by setting the whole object structure to 0?